tools/c7n_tencentcloud/c7n_tencentcloud/actions/tags.py

```diff
@@ -1,11 +1,305 @@
+        self.log = logging.getLogger("custodian.tencentcloud.actions.TagAction")
+        self.batch_size = 10
+        self.tags = []
+
+    def process(self, resources):
+        self.process_tag_op(resources)
+
+    def _get_request_params(self, resources, tags):
+        """
+        _get_request_params
+        The tag request parameter, the default is to add the tag parameter,
+        if the action is UnTagResources, it means to delete the tag
+        """
+        qcs_list = self.manager.source.get_resource_qcs(resources)
+
+        if self.action == "UnTagResources":
+            return {"ResourceList": qcs_list, "TagKeys": tags}
+        else:
+            return {"ResourceList": qcs_list, "Tags": tags}
+
+    def process_tag_op(self, resources):
+        """process_tag"""
+        try:
+            client = self.get_client()
+            for batch in chunks(resources, self.batch_size):
+                params = self._get_request_params(batch, self.tags)
+                resp = client.execute_query(self.action, params)
+                failed_resources = jmespath.search("Response.FailedResources[]", resp)
+                params_str = json.dumps(params)
+                if len(failed_resources) != 0:
+                    self.log.error("operation failed. %s , params: %s", self.data.get('type'),
+                                   params_str)
+                self.log.debug("%s , params: %s ", self.data.get('type'), params_str)
+            return True
+        except (RetryError, TencentCloudSDKException) as err:
+            raise PolicyExecutionError(err) from err
+
+
+class AddTagAction(TagAction):
+    """
+    Add tag information
+    """
+    schema = type_schema("tag",
+                         key={"type": "string"},
+                         value={"type": "string"})
+    schema_alias = True
+
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super().__init__(data, manager, log_dir)
+        self.action = "TagResources"
+        self.tags = [{"TagKey": self.data.get("key"), "TagValue": self.data.get("value")}]
+
+    def validate(self):
+        """validate"""
+        if not self.data.get('key') or not self.data.get('value'):
+            raise PolicyValidationError("Must specify key")
+        return self
+
+
+class RenameTagAction(TagAction):
+    """
+    Rename the tag information, because Tencent Cloud API does not support direct modification,
+    you need to delete it first and then add it
+    """
+    schema = type_schema("rename-tag",
+                         old_key={"type": "string"},
+                         new_key={"type": "string"})
+    schema_alias = True
+
+    def validate(self):
+        """validate"""
+        if not self.data.get('old_key') or not self.data.get('new_key'):
+            raise PolicyValidationError("Must specify key")
+        return self
+
+    def _get_tag_params(self, resources):
+        old_key = self.data.get('old_key')
+        tags = resources[0]["Tags"]
```

Comment:
why are tag params for the set being calculated from a single resource?

Reply:
Actually it was an issue here. 
We use another API now, which is more suitable for this case.
Also we have updated the test cases to cover multi resources  situation.